runtime/server_http_request.go

```diff
@@ -516,148 +516,161 @@ func (req *ServerHTTPRequest) GetQueryValueList(key string) ([]string, bool) {
 
 // -- Query param as set --
 
+/**
+ * A set of bools does not make sense and is unimplemented
+ * Also, in every use-case for a gateway, a set implemented as a map is not very useful, instead one where
+ * it is a list with no duplicates is. Therefore the implementation picks that approach.
+ */
+
 // The "value" in the map representation of a set datastructure
 var _nullVal = struct{}{}
 
-// GetQueryBoolSet will return a query param as a set of boolean
-// @argo: Does this method even make sense?
-func (req *ServerHTTPRequest) GetQueryBoolSet(key string) (map[bool]struct{}, bool) {
-	success := req.parseQueryValues()
-	if !success {
-		return nil, false
-	}
-
-	values := req.queryValues[key]
-	ret := make(map[bool]struct{}, len(values))
-	for _, value := range values {
-		if value == "true" {
-			ret[true] = _nullVal
-		} else if value == "false" {
-			ret[false] = _nullVal
-		} else {
-			err := &strconv.NumError{
-				Func: "ParseBool",
-				Num:  value,
-				Err:  strconv.ErrSyntax,
-			}
-			req.LogAndSendQueryError(err, "bool", key, value)
-			return nil, false
-		}
-	}
-
-	return ret, true
-}
-
-// GetQueryInt8Set will return a query params as set of int8
-func (req *ServerHTTPRequest) GetQueryInt8Set(key string) (map[int8]struct{}, bool) {
+// GetQueryInt8Set will return a query params as set of int8 (implemented as a deduped slice)
+func (req *ServerHTTPRequest) GetQueryInt8Set(key string) ([]int8, bool) {
 	success := req.parseQueryValues()
 	if !success {
 		return nil, false
 	}
 
 	values := req.queryValues[key]
-	ret := make(map[int8]struct{}, len(values))
+	set := make(map[int8]struct{}, len(values))
 	for _, value := range values {
-		number, err := strconv.ParseInt(value, 10, 8)
+		number, err := strconv.ParseInt(value, 0, 8)
 		if err != nil {
 			req.LogAndSendQueryError(err, "int8", key, value)
 			return nil, false
 		}
-		ret[int8(number)] = _nullVal
+		set[int8(number)] = _nullVal
```

Comment:
why create a separate typedef for `struct{}{}`. In go, it's quite standard to use empty struct as value when you want to use map as a set but now one has to look up what is `_nullVal` to understand you actually want a set.

Reply:
I did it last time because the struct{} was not well known to me -- and a stack-overflow question had that kind of thing.  No longer needed.  I just let the thing be - with minimal changes.